data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,131 @@
+# Kafka source
+
+This is the Data Prepper Kafka source plugin that reads records from Kafka topic. It uses the consumer API provided by Kafka to read messages from the broker.
+
+The Kafka source plugin supports OpenSearch 2.0.0 and greater.
+
+## Usages
+
+The Kafka source should be configured as part of Data Prepper pipeline yaml file.
+
+## Configuration Options
+
+```
+log-pipeline:
+  source:
+    kafka:
+      bootstrap_servers:
+        - 127.0.0.1:9092,localhost:9093, 127.0.0.1:9094
+      topics:
+        - topic:
+            name: my-topic-2
+            auth_type: SASL_OAUTH #optional
+            consumer:
+              group_name: kafka-consumer-group-2
+              group_id: DPKafkaProj-2
```

Comment:
We should try to avoid creating unnecessary nesting in the YAML. I understand that the "connection" concept is logically connected. But some of these are required.

So, we are making users add this `consumer` group for even the simplest use-cases.

I'd like for a minimal pipeline configuration to not require any additional nesting. See this example:

```
log-pipeline:
  source:
    kafka:
      boostrap_servers: ["127.0.0.1:9020"]
      group_id: DPKafkaProj-2
      topics: [ "my-topic" ]
```